python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+        and connection for image metadata.
+
+        Parameters
+        ----------
+        butlerget : locateImage.ButlerGet
+            the butler instance and config info.
+        metaservget : locateImage.MetaservGet
+            provides access to image metadata.
+        logger : log
+            the logger to be used.
+        """
+        self._log = logger
+        self._butler = butlerget.butler
+        self._butler_keys = sorted(butlerget.butler_keys)
+        self._imagedataset_type = butlerget.butler_policy
+        self._metaservget = metaservget
+
+    def full_nearest(self, ra, dec, filt):
+        """Returns image containing center(x,y) of unit and filter.
+
+        Parameters
+        ----------
+        ra : `float`
+            x-coordinate, or Ra if Equatorial
+        dec : `float`
+            y-coordinate, or Dec if Equatorial
+        filt(optional): `str`
+            ['u', 'g', 'r', 'i', 'z', 'y']
+
+        Returns
+        -------
+        image : `afwImage.Exposure`
+
+        """
+        q_result = self._metaservget.nearest_image_containing(ra, dec, filt)
+        if not q_result:
+            # not found
+            return None
+        else:
+            data_id = self._data_id_from_qr(q_result)
+            image = self._image_from_butler(data_id)
+            return image
+
+    def full_from_data_id_by_run(self, run, camcol, field, filt):
+        """Returns image from specified data id (run, camcol, field, filter).
+
+        Parameters
+        ----------
+        run : `int`
+        camcol : `int`
+        field : `int`
+        filt : `str`
+            ['u', 'g', 'r', 'i', 'z', 'y']
+
+        Returns
+        -------
+        image: `afwImage.Exposure`
+
+        """
+        image = self._butler.get(self._imagedataset_type, run=run,
+                                 camcol=camcol, field=field, filter=filt)
+        return image
+
+    def full_from_data_id_by_tract(self, tract, patch_x, patch_y, filt):
+        """Returns image from specified data id (tract, patch<x,y>, filter).
+
+        Parameters
+        ----------
+        tract : `int`
+        patch_x : `int`
+        patch_y : `int`
+        filt : `str`
+            ['u', 'g', 'r', 'i', 'z', 'y']
+
+        Returns
+        -------
+        image : `afwImage.Exposure`
+
+        """
+        patch = ",".join((str(patch_x), str(patch_y)))
+        image = self._butler.get(self._imagedataset_type, tract=tract,
+                                 patch=patch, filter=filt)
+        return image
+
+    def full_from_ccd_exp_id(self, ccd_exp_id):
+        """Returns image from the science id.
+
+        Parameters
+        ----------
+        ccd_exp_id : `int`
+
+        Returns
+        -------
+        image: `afwImage.Expsoure`
+
+        """
+        data_id = self.data_id_from_ccd_exp_id(ccd_exp_id)
+        if data_id:
+            image = self._image_from_butler(data_id)
+            return image
```

Comment:
Again you might think about throwing an exception here rather than returning None and having to check it somewhere else (and likely throw an exception there).